nirodents/workflows/brainextraction.py

```diff
@@ -53,121 +59,193 @@ def init_rodent_brain_extraction_wf(
-    )
+    if not tpl_target_path:
+        raise RuntimeError(
+            f"An instance of template <tpl-{in_template}> with MR scheme '{bids_suffix}'"
+            " could not be found.")
 
-    # resample template and target
-    res_tmpl = pe.Node(
-        RegridToZooms(in_file=tpl_target_path, zooms=LOWRES_ZOOMS), name="res_tmpl"
+    tpl_brainmask_path = get_template(
+        in_template, resolution=2, atlas=None, desc="brain", suffix="probseg",
     )
+    if not tpl_brainmask_path:
+        tpl_brainmask_path = get_template(
+            in_template, resolution=2, atlas=None, desc="brain", suffix="mask",
+        )
 
-    res_target = pe.Node(RegridToZooms(zooms=LOWRES_ZOOMS), name="res_target")
-    res_target2 = pe.Node(RegridToZooms(zooms=LOWRES_ZOOMS), name="res_target2")
+    # Resample both target and template to a controlled, isotropic resolution
+    res_tmpl = pe.Node(RegridToZooms(zooms=HIRES_ZOOMS), name="res_tmpl")
+    res_target = pe.Node(RegridToZooms(zooms=HIRES_ZOOMS), name="res_target")
+    lowres_tmpl = pe.Node(RegridToZooms(zooms=LOWRES_ZOOMS), name="lowres_tmpl")
+    lowres_target = pe.Node(RegridToZooms(zooms=LOWRES_ZOOMS), name="lowres_target")
+    gauss_tmpl = pe.Node(niu.Function(function=_gauss_filter), name="gauss_tmpl")
+    gauss_tmpl.inputs.sigma = tuple(np.array(LOWRES_ZOOMS) * 8.0)
 
-    dil_mask = pe.Node(
-        MaskTool(outputtype="NIFTI_GZ", dilate_inputs="2", fill_holes=True),
-        name="dil_mask",
+    # Initialize transforms with antsAI
+    init_aff = pe.Node(
+        AI(
+            metric=("Mattes", 32, "Regular", 1.0),
+            transform=("Affine", 0.1),
+            search_factor=(10, 0.08),
+            principal_axes=False,
+            convergence=(40, 1e-6, 10),
+            search_grid=(25, (0, 0, 0)),
+            verbose=True,
+        ),
+        name="init_aff",
+        n_procs=omp_nthreads,
     )
 
-    # truncate target intensity for N4 correction
-    trunc_opts = {"T1w": "0.005 0.999 256", "T2w": "0.01 0.999 256"}
-    trunc = pe.MapNode(
-        ImageMath(operation="TruncateImageIntensity", op2=trunc_opts[bids_suffix]),
-        name="truncate_images",
-        iterfield=["op1"],
-    )
+    # main workflow
+    wf = pe.Workflow(name)
+    # Create a buffer interface as a cache for the actual inputs to registration
+    buffernode = pe.Node(niu.IdentityInterface(
+        fields=["lowres_target", "hires_target"]), name="buffernode")
 
-    # Initial N4 correction
-    inu_n4 = pe.MapNode(
-        N4BiasFieldCorrection(
-            dimension=3,
-            save_bias=False,
-            copy_header=True,
-            n_iterations=[50] * 4,
-            convergence_threshold=1e-7,
-            shrink_factor=4,
-            bspline_fitting_distance=bspline_fitting_distance,
-        ),
-        n_procs=omp_nthreads,
-        name="inu_n4",
-        iterfield=["input_image"],
+    if bids_suffix.lower() == "t2w":
```

Comment:
I have only addressed the t2 path

Reply:
The mp2rage is much simpler, since the resampled image can go straight into `antsAI`